### PR TITLE
Add GCS Action

### LINKED ISSUE #32: Feature request: actions for working with cloud storage
In my projects, I often have to build and publish documentation as static websites. I typically host these files with Cloud Storage, so having an easy to use action that takes a folder in my repository and uploads it to a given bucket.

Perhaps a simplified workflow could be something like:

```yaml
name: Master workflow

on:
  push:
    branches:
      - master

jobs:
  build-and-publish-docs:
    name: Build and publish documentation
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v1
      - name: Build documentation
        run: |
          do whatever to build the documentation
      - uses: ____/publish-to-gcs@v666
        with:
          source-folder: ./generated/docs/of/doom
          bucket-name: my-amazeballs-bucket
          object-key-prefix: /the-docs/1.0.1/
          clear-existing-files-first: true
```

### PR DESCRIPTION
**Pull request by bharathkkb**
replaces #55 
fixes #32 
Add a native GCS Upload Action using google-cloud/storage

### CODE REVIEW

File: upload-cloud-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+# upload-cloud-storage
+
+This action uploads files/folders to a [Google Cloud Storage (GCS)][gcs] bucket. This is useful when
+you want upload build artifacts from your workflow.
+
+Paths to files that are successfully uploaded are set as output variables and can be
+used in subsequent steps.
+
+## Prerequisites
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply upload-cloud-storage/README.md:25]: Storage(storage.googleapis.com) is usually enabled by default, not sure if we should call that out.
https://cloud.google.com/service-usage/docs/enabled-service#default

File: upload-cloud-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+permissions to access the secrets being requested.
+
+### Via the setup-gcloud action
+
+You can provide credentials using the [setup-gcloud][setup-gcloud] action:
+
+```yaml
+- uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+  with:
+    export_default_credentials: true
```
[the remark this answers is not in the document]
Comment (bharathkkb) [reply upload-cloud-storage/README.md:152]: added

**Comment by averikitsch** [comment]
ping. Do you want to go ahead and merge this?

**Comment by bharathkkb** [comment]
@averikitsch sorry missed Craig's approval. I have merged it.

**Pull request merged**